### PR TITLE
Adding the FontFactory class

### PR DESCRIPTION
**Pull request by iifawzi**
### Description

This pull request added the `FontFactory` class (the `customFont` class before)
The changes: 
- Combined the `gruntCdIcons` function with the `commandFinal` function. 
- Refactored the `generateFiles` function to use the `zipFolder` and `addConfigFiles` functions which we introduced before. 
- Renamed the class. 

- I've added a `tempDirectory` and `svgDirectory` constants, we're using these constants in multiple files.

### CODE REVIEW

File: src/utils/FontFactory.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      try {
+        let command = ''
+        for (let i = 0; i < this.icons.length; i++) {
+          command += `--extended_src=${this.theme}/${this.icons[i]}.svg `
+        }
+        return `grunt -b ./src --dist=${this.timestamp} ${command} --outlined=${this.theme === 'svg-outlined'}`
+      } catch (err) {
+        throw new Error(`Some error occurred while generating Icons set command: ${err}`)
+      }
+    }
```
Comment (kaybhutani) [inline src/utils/FontFactory.ts:30]: the `gruntCdIcons()` generates the command and `generateFIles` actually doesnt generate files, it is just zipping the already generated files. as far as i remember we had a package node-cli to run the grunt command, can we move that functions here to to this class itself?

File: src/common/constants.ts [hunk 2]
```diff
@@ -8,3 +10,10 @@ export const FILES_PATHS = {
   OUTLINED_TEMP: 'svg-outlined-temp/',
   FILLED_TEMP: 'svg-temp/'
 }
+
+// *********************************
+// Directories used in APIs (V1)
+// *********************************
+
+export const tempDirectory = path.join(__dirname, '..', '/temp/')
```
Comment (kaybhutani) [inline src/common/constants.ts]: lets remove the 2nd slash and keep them like `/temp`. The second slash should come from sub directories, just like routes. Earlier we had a few bugs where in some places the parent directory had the `/` and some places it doesnt :p
so lets be consistent

File: src/utils/FontFactory.ts (hunk 1 above)
Comment (iifawzi) [reply src/utils/FontFactory.ts:30]: Yes, It make more sense, to be moved to the class itself. I will work on it

**Comment by kaybhutani** [comment]
Looks good

**Pull request merged**